src/bp/nlu-core/engine.ts

```diff
@@ -184,6 +183,21 @@ export default class Engine implements NLU.Engine {
     return Engine._trainingWorkerQueue.cancelTraining(trainSessionId)
   }
 
+  private _mergeModelOutputs(
+    currentOutput: TrainOutput,
+    previousOutput: TrainOutput,
+    allContexts: string[]
+  ): TrainOutput {
+    const output = { ...currentOutput }
+
+    const previousIntents = _.pick(previousOutput.intent_model_by_ctx, allContexts)
+    const previousOOS = _.pick(previousOutput.oos_model, allContexts)
+
+    output.intent_model_by_ctx = { ...previousIntents, ...currentOutput.intent_model_by_ctx }
+    output.oos_model = { ...previousOOS, ...currentOutput.oos_model }
```

Comment:
We'll still have the entity cache dump messed up because of key ordering am I right ?

Reply:
Nope, the cache is not inside `intent_model_by_ctx` or `oos_model`

\+ spread operator does a shallow merge, not a deep merge like lodash:


![image](https://user-images.githubusercontent.com/25970722/93228224-458d9e80-f743-11ea-863d-593b541901ef.png)